pkg/user/parallel_tx_submission.go

```diff
@@ -0,0 +1,497 @@
+		p.initialized.Store(true)
+	}
+
+	// Create a new context for this pool instance
+	p.ctx, p.cancel = context.WithCancel(ctx)
+
+	for _, worker := range p.workers {
+		go worker.start(p.ctx)
+	}
+
+	p.started.Store(true)
+	return nil
+}
+
+// stop shuts down all workers in the pool
+func (p *txQueue) stop() {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	if !p.started.Load() {
+		return
+	}
+
+	if p.cancel != nil {
+		p.cancel()
+	}
+	p.started.Store(false)
+}
+
+// submitJob submits a job to the parallel worker pool
+func (p *txQueue) submitJob(job *SubmissionJob) {
+	p.mtx.RLock()
+	jobQueue := p.jobQueue
+	ctx := p.ctx
+	started := p.started.Load()
+	p.mtx.RUnlock()
+
+	if !started || ctx == nil {
+		job.ResultsC <- SubmissionResult{Error: errors.New("tx queue not started")}
+		return
+	}
+
+	select {
+	case jobQueue <- job:
+	case <-ctx.Done():
+		job.ResultsC <- SubmissionResult{Error: errors.New("tx queue full or has stopped")}
+	}
+}
+
+// isStarted returns whether the tx queue is started
+func (p *txQueue) isStarted() bool {
+	return p.started.Load()
+}
+
+// start begins the worker's job processing loop
+func (w *txWorker) start(ctx context.Context) {
+	for {
+		select {
+		case job := <-w.jobQueue:
+			w.processJob(job, ctx)
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+// processJob handles the actual transaction submission for a job
+func (w *txWorker) processJob(job *SubmissionJob, workerCtx context.Context) {
+	jobCtx := job.Ctx
+	if jobCtx == nil {
+		jobCtx = context.Background()
+	}
+
+	select {
+	case <-jobCtx.Done():
+		job.ResultsC <- SubmissionResult{Signer: w.address, Error: jobCtx.Err()}
+		return
+	case <-workerCtx.Done():
+		job.ResultsC <- SubmissionResult{Signer: w.address, Error: workerCtx.Err()}
+		return
+	default:
+	}
+
+	options := job.Options
+
+	// Only add fee granter option for workers that aren't the primary account (worker 0)
+	if w.id != 0 {
+		// Add fee granter option so master account pays for worker transaction fees
+		options = append([]TxOption{SetFeeGranter(w.client.DefaultAddress())}, options...)
+	}
+
+	// Use the worker's dedicated account to submit the transaction
+	txResponse, err := w.client.SubmitPayForBlobWithAccount(jobCtx, w.accountName, job.Blobs, options...)
+
+	result := SubmissionResult{
+		Signer:     w.address,
+		TxResponse: txResponse,
+		Error:      err,
+	}
+
+	// Send result back through the job-specific results channel
+	job.ResultsC <- result
+}
+
+// InitializeWorkerAccounts creates and initializes all worker accounts for parallel submission.
+// It creates the accounts in the keyring if they don't exist, funds them with a small balance,
+// and sets up fee grants so the main account pays for transaction fees.
+// This method should be called after TxClient creation but before parallel submissions.
+func (client *TxClient) InitializeWorkerAccounts(ctx context.Context) error {
+	if client.txQueue == nil {
+		return errors.New("tx queue not configured - use WithTxWorkers option")
```

Comment:
return errors.New("tx queue not configured - use WithTxWorkers option")
```suggestion
		return errors.New("tx queue not configured, use WithTxWorkers option")
```